File: examples/python/extract.py
"""Extract plain text and Markdown from an Office document."""
import sys
from office_oxide import Document


def main(path: str) -> None:
    with Document.open(path) as doc:
        print(f"format: {doc.format}")
        print("--- plain text ---")
        print(doc.plain_text())
        print("--- markdown ---")
        print(doc.to_markdown())


if __name__ == "__main__":
    if len(sys.argv) != 2:
        sys.exit("usage: extract.py <file>")
    main(sys.argv[1])

File: examples/javascript/extract.mjs
#!/usr/bin/env node
import { Document } from 'office-oxide';

if (process.argv.length !== 3) {
  console.error('usage: extract.mjs <file>');
  process.exit(1);
}

const doc = Document.open(process.argv[2]);
try {
  console.log('format:', doc.format);
  console.log('--- plain text ---');
  console.log(doc.plainText());
  console.log('--- markdown ---');
  console.log(doc.toMarkdown());
} finally {
  doc.close();
}
